Sources/Overlays/_Testing_Foundation/Attachments/Attachable+NSSecureCoding.swift

```diff
@@ -0,0 +1,78 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+#if canImport(Foundation)
+@_spi(Experimental) public import Testing
+public import Foundation
+
+// As with Encodable, implement the protocol requirements for
+// NSSecureCoding-conformant classes by default. The implementation uses
+// NSKeyedArchiver for encoding.
+@_spi(Experimental)
+extension Attachable where Self: NSSecureCoding {
+  /// Encode this object using [`NSKeyedArchiver`](https://developer.apple.com/documentation/foundation/nskeyedarchiver)
+  /// into a buffer, then call a function and pass that buffer to it.
+  ///
+  /// - Parameters:
+  ///   - attachment: The attachment that is requesting a buffer (that is, the
+  ///     attachment containing this instance.)
+  ///   - body: A function to call. A temporary buffer containing a data
+  ///     representation of this instance is passed to it.
+  ///
+  /// - Returns: Whatever is returned by `body`.
+  ///
+  /// - Throws: Whatever is thrown by `body`, or any error that prevented the
+  ///   creation of the buffer.
+  ///
+  /// The testing library uses this function when writing an attachment to a
+  /// test report or to a file on disk. The encoding used depends on the path
+  /// extension specified by the value of `attachment`'s ``Testing/Attachment/preferredName``
+  /// property:
+  ///
+  /// | Extension | Encoding Used | Encoder Used |
+  /// |-|-|-|
+  /// | `".xml"` | XML property list | [`NSKeyedArchiver`](https://developer.apple.com/documentation/foundation/nskeyedarchiver) |
+  /// | None, `".plist"` | Binary property list | [`NSKeyedArchiver`](https://developer.apple.com/documentation/foundation/nskeyedarchiver) |
```

Comment:
Not sure whether it's correct per se, but I've certainly seen a lot of XML property lists which use the `.plist` file extension. From a user's perspective, to make inspecting the contents easier with my chosen text editor I definitely prefer XML. (Of course, binary property lists have their place too—primarily when meant for machine consumption.)

I'm wondering whether we really _need_ to have the `.plist` extension encode as binary, or whether it could use XML too? I'm trying to optimize this for consumption by human test authors. I realize you can just switch to `.xml` as a workaround, but it seems easily to forget or overlook, and people may prefer `.plist`.

Reply:
This choice is intentional as there is no other way to specify which subformat of property list you want. A future PR will introduce a `contentType` property on Apple platforms (the non-Apple platform situation is TBD) that we would then use to refine this logic.

Reply:
Ok great